F21/annoyed-raven/main.py

```diff
@@ -0,0 +1,104 @@
+import csv
+
+capitalized = []
+lowercased = []
+language_encoding = []
+
+'''
+This function opens the tsv file and seperates tests cases into three parts
+the language encoding, Capitilazed input, and the suggested Lowercased result
+'''
+with open("tests.tsv", encoding="UTF-8") as language_tsv:
+    language_tsv = csv.reader(language_tsv, delimiter="\t", quotechar='"')
+    for row in language_tsv:
+        transition = []
+        for x in row:
+            transition.append(x)
+         
+        capitalized.append(transition[0])
+        language_encoding.append(transition[1])
+        lowercased.append(transition[2])       
+'''     an easy check to make sure tsv file was seperated correctly
+        print(row)
+        print(capitalized)
+        print(language_encoding)
+        print(lowercased)
+'''  
+# the storage lists of indexes for each occurence of a the specific dialact/language
+Encoding_indexesEN= []
+Encoding_indexesEN_US= []
+Encoding_indexesEN_IE= []
+Encoding_indexesEN_LATN= []
+Encoding_indexesGA= []
+Encoding_indexesGA_IE= []
+Encoding_indexesTR= []
+Encoding_indexesEL= []
+Encoding_indexesZH_HANS= []
+Encoding_indexesTH= []
+
+#sorting of each occurance per dialect/language
+for i in range(0, len(language_encoding)):
+    if language_encoding[i] == "en":
+        Encoding_indexesEN.append(i)
+    elif language_encoding[i] == "en-US":
+        Encoding_indexesEN_US.append(i)    
+    elif language_encoding[i] == "en-IE":
+        Encoding_indexesEN_IE.append(i)  
+    elif language_encoding[i] == "en-Latn":
+        Encoding_indexesEN_LATN.append(i)  
+    elif language_encoding[i] == "ga":
+        Encoding_indexesGA.append(i)  
+    elif language_encoding[i] == "ga-IE":
+        Encoding_indexesGA_IE.append(i)  
+    elif language_encoding[i] == "tr":
+        Encoding_indexesTR.append(i)  
+    elif language_encoding[i] == "el":
+        Encoding_indexesEL.append(i)  
+    elif language_encoding[i] == "zh-Hans":
+        Encoding_indexesZH_HANS.append(i)  
+    elif language_encoding[i] == "th":
+        Encoding_indexesTH.append(i)  
+
+
+# final sorting and printing of words
+# still missing exceptions
+for j in Encoding_indexesEN:
+    instance_en = []
+    instance_en.append(capitalized[j].lower())
+    print(instance_en)
+for j in Encoding_indexesEN_US:
+    instance_en_us = []
+    instance_en_us.append(capitalized[j].lower())
+    print(instance_en_us)
+for j in Encoding_indexesEN_IE:
+    instance_en_ie = []
+    instance_en_ie.append(capitalized[j].lower())
+    print(instance_en_ie)
+for j in Encoding_indexesEN_LATN:
+    instance_en_latn = []
+    instance_en_latn.append(capitalized[j].lower())
+    print(instance_en_latn)
+for j in Encoding_indexesGA:
```

Comment:
The standard Python lower() method will not always work for Gaelic as it cannot properly handle diacritic characters. There are several ways to handle it though I am only familiar with one through experience. The method I am familiar with was rather involved as I had to essentially build a miniature library in order to properly handle them